src/xkit.cs

```diff
@@ -1190,4 +1190,87 @@ interface NSLayoutManagerDelegate {
 		[Export ("layoutManager:shouldSetLineFragmentRect:lineFragmentUsedRect:baselineOffset:inTextContainer:forGlyphRange:")]
 		bool ShouldSetLineFragmentRect (NSLayoutManager layoutManager, ref CGRect lineFragmentRect, ref CGRect lineFragmentUsedRect, ref nfloat baselineOffset, NSTextContainer textContainer, NSRange glyphRange);
 	}
+
+	[NoWatch, TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	interface NSDiffableDataSourceSnapshot<SectionIdentifierType, ItemIdentifierType> : NSCopying
+		where SectionIdentifierType : NSObject
+		where ItemIdentifierType : NSObject {
+
+		[Export ("numberOfItems")]
+		nint NumberOfItems { get; }
+
+		[Export ("numberOfSections")]
+		nint NumberOfSections { get; }
+
+		[Export ("sectionIdentifiers")]
+		SectionIdentifierType [] SectionIdentifiers { get; }
+
+		[Export ("itemIdentifiers")]
+		ItemIdentifierType [] ItemIdentifiers { get; }
+
+		[Export ("numberOfItemsInSection:")]
+		nint GetNumberOfItems (SectionIdentifierType sectionIdentifier);
+
+		[Export ("itemIdentifiersInSectionWithIdentifier:")]
+		ItemIdentifierType [] GetItemIdentifiersInSection (SectionIdentifierType sectionIdentifier);
+
+		[Export ("sectionIdentifierForSectionContainingItemIdentifier:")]
+		[return: NullAllowed]
+		SectionIdentifierType GetSectionIdentifierForSection (ItemIdentifierType itemIdentifier);
+
+		[Export ("indexOfItemIdentifier:")]
+		nint GetIndex (ItemIdentifierType itemIdentifier);
+
+		[Export ("indexOfSectionIdentifier:")]
+		nint GetIndex (SectionIdentifierType sectionIdentifier);
+
+		[Export ("appendItemsWithIdentifiers:")]
+		void AppendItems (ItemIdentifierType [] identifiers);
+
+		[Export ("appendItemsWithIdentifiers:intoSectionWithIdentifier:")]
+		void AppendItems (ItemIdentifierType [] identifiers, SectionIdentifierType sectionIdentifier);
```

Comment:
Same, Append?

Reply:
too late anyway, that has already shipped (this is code being moved from files)